include/graybat/utils/MultiKeyMap.hpp

```diff
@@ -33,214 +33,147 @@
 // BOOST
 #include <boost/optional.hpp>
 
-// HANA
-#include <boost/hana.hpp>
-namespace hana = boost::hana;
-
 namespace utils {
-
-	template <typename... Tail> struct MultiKeyMapType;
-
-	template <typename Key, typename Value>
-	struct MultiKeyMapType<Key, Value> {
-		typedef std::map<Key, Value> type;
+	template <class... Types>
+	struct Difference;
+
+	template <
+		class T,
+		class... Types,
+		class... subTypes
+	>
+	struct Difference<std::tuple<T, Types...>, std::tuple<T, subTypes...>> {
+		using ValueType = typename Difference<
+			std::tuple<Types...>, 
+			std::tuple<subTypes...>
+		>::ValueType;
```

Comment:
Could you provide a small javadoc style documentation for this struct e.g.: brief description, precondition, postcondition, invariant, and a small usage example.